File: src/main/java/io/github/mdsimmo/bomberman/events/BmIntent.java
package io.github.mdsimmo.bomberman.events;

public abstract class BmIntent extends BmEvent {

    private boolean handled;

    public void setHandled() {
        this.handled = true;
    }

    public boolean isHandled() {
        return handled;
    }

    public void verifyHandled() {
        if (!handled)
            throw new RuntimeException("Event not handled: " + this);
    }

}

File: src/main/java/io/github/mdsimmo/bomberman/events/BmIntent.kt
package io.github.mdsimmo.bomberman.events

class BmIntent : Intent {

    private var isHandled = false

    override fun isHandled(): Boolean = isHandled

    override fun setHandled() {
        isHandled = true
    }

    override fun verifyHandled() {
        if (!isHandled) throw RuntimeException("Event not handled: $this")
    }
}